### DISCUSSION TITLE
Why the level shifter?

### DISCUSSION THREAD

**Discussion by crazynightgriffin** [visualapproach/WiFi-remote-for-Bestway-Lay-Z-SPA#456] [General]
As I am waiting for my parts to arrive, I was searching around and found this blog, uses the code from this Github but no level shifter....  

https://antanaitis.lt/2022/09/07/wi-fi-on-your-lay-z-spa-make-it-easy-to-remote-control/

What are peoples thoughts?

**Comment by visualapproach** [comment]
#312

**Reply by visualapproach** [reply]
ESP uses 3.3V. Pump 5V. The ESP is 5V tolerant and if you're lucky, the 3.3V is enough to drive the 5V signal high. And you probably are.

**Reply by crazynightgriffin** [reply]
Thanks for the explanation I do understand a little about signal voltage etc hence my question so in an ideal world the level shifter is needed but things may/should work if lucky but essentially not blow anything up just not provide enough to send the pump high.

**Reply by visualapproach** [reply]
Exactly :-)

**Reply by crazynightgriffin** [reply]
Off to the shed to look for a Wemos D1 :)

**Reply by booyakafoo** [reply]
Did you try it out?  What were your results?

**Reply by crazynightgriffin** [reply]
> Did you try it out? What were your results?

Yep working perfectly, now able to heat to 40'c over night then drop to 38'c, today spotted the batteries at 100% and was ale to knock it back to 40'c for free :)

**Reply by booyakafoo** [reply]
I didn't wait to see and tried it myself.  Got it working as well.  Integrated into home assistant also.  Was a fun little project.  Thanks for sharing the link above.  It made this less intimidating.
